### PR TITLE
WFLY-15546 Move WildFly Preview to a native Jakarta namespace variant of the jpa subsystem module

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.redhat.com/browse/WFLY-15546

Signed-off-by: Scott Marlow <smarlow@redhat.com>

### CODE REVIEW

File: ee-9/source-transform/pom.xml [hunk 1]
```diff
@@ -49,6 +49,7 @@
         <module>ee</module>
         <module>ee-security</module>
         <module>jpa/spi</module>
+	<module>jpa/subsystem</module>
```
Comment (yersan) [inline ee-9/source-transform/pom.xml:52]: Minor formatting issue

**Review by yersan — CHANGES_REQUESTED** [review]
Hello @scottmarlow , looking at the "Wildfly Preview - Linux - JDK 17" traces, the wildfly-jpa subsystem is still being transformed there:

`[19:42:41] :	 [Step 2/5] [INFO] EE9: transformed wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar`

There is still something in the subsystem which is being transformed. I had similar issues when I worked on JSF, take a look at this comment to know how you can check what's pending:

https://github.com/wildfly/wildfly/pull/14823#issuecomment-948212063

**Comment by scottmarlow** [comment]
> Hello @scottmarlow , looking at the "Wildfly Preview - Linux - JDK 17" traces, the wildfly-jpa subsystem is still being transformed there:
> 
> `[19:42:41] : [Step 2/5] [INFO] EE9: transformed wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar`
> 
> There is still something in the subsystem which is being transformed. I had similar issues when I worked on JSF, take a look at this comment to know how you can check what's pending:
> 
> [#14823 (comment)](https://github.com/wildfly/wildfly/pull/14823#issuecomment-948212063)


I do not see a comment at https://github.com/wildfly/wildfly/blob/main/ee-9/feature-pack/pom.xml#L322.

Was it the following line that you uncommented?
```
 <!--<jakarta-transform-verbose>true</jakarta-transform-verbose>-->
```

**Comment by yersan** [comment]
Ups, yes, this code is changing very quickly, yes uncomment the ` <!--<jakarta-transform-verbose>true</jakarta-transform-verbose>-->` to get more information about why an artifact is being transformed

**Comment by bstansberry** [comment]
@scottmarlow https://gist.github.com/bstansberry/5df0cbd64ae3b35c6a60964b00058f0b shows the details about why wildfly-jpa-jakarta is being transformed.

**Comment by scottmarlow** [comment]
> @scottmarlow https://gist.github.com/bstansberry/5df0cbd64ae3b35c6a60964b00058f0b shows the details about why wildfly-jpa-jakarta is being transformed.

Some of what I see:

```
[INFO] Stop processing [ org/jboss/as/jpa/service/PersistenceUnitServiceImpl$1$1.class ] using [ CLASS ]: Content changes
[INFO] Stop processing [ org/jboss/as/jpa/service/PhaseOnePersistenceUnitServiceImpl.class ] using [ CLASS ]: Content changes
[INFO] Stop processing [ org/jboss/as/jpa/service/PhaseOnePersistenceUnitServiceImpl$2$1.class ] using [ CLASS ]: Content changes
[INFO] Stop processing [ org/jboss/as/jpa/service/PhaseOnePersistenceUnitServiceImpl$1$1.class ] using [ CLASS ]: Content changes
[INFO] Stop processing [ org/jboss/as/jpa/service/PersistenceUnitServiceImpl.class ] using [ CLASS ]: Content changes
```

And:

```
Stop processing [ /Users/bstansberry/.m2/repository/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ] using [ JAR ]: Content changes
[INFO] ================================================================================
[INFO] [ Input  ] [ /Users/bstansberry/.m2/repository/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ]
[INFO]            [ /Users/bstansberry/.m2/repository/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ]
[INFO] [ Output ] [ /Users/bstansberry/.m2/repository/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ]
[INFO]            [ /Users/bstansberry/dev/wildfly/wf2/wildfly/ee-9/feature-pack/target/jakarta-transform-maven-repo/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ]
[INFO] ================================================================================
[INFO] [ Immediate changes: ]
[INFO] --------------------------------------------------------------------------------
[INFO] [          All Resources ] [    172 ] Unselected [      0 ] Selected [    172 ]
[INFO] --------------------------------------------------------------------------------
[INFO] [            All Actions ] [    172 ]  Unchanged [    167 ]  Changed [      5 ]
[INFO] [           Class Action ] [    134 ]  Unchanged [    129 ]  Changed [      5 ]
[INFO] [ Properties file relocate ] [      2 ]  Unchanged [      2 ]  Changed [      0 ]
[INFO] [        Manifest Action ] [      1 ]  Unchanged [      1 ]  Changed [      0 ]
[INFO] [  Service Config Action ] [      2 ]  Unchanged [      2 ]  Changed [      0 ]
[INFO] [            Text Action ] [      4 ]  Unchanged [      4 ]  Changed [      0 ]
[INFO] [            Null Action ] [     29 ]  Unchanged [     29 ]  Changed [      0 ]
[INFO] ================================================================================
[INFO] EE9: transformed wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar
@scottmarlow
```

So to complete the transformation of `wildfly-jpa` into `wildfly-jpa-jakarta`, we need additional changes to also transform the identified five class files for the identified constant pool entries.

**Comment by bstansberry** [comment]
@ropalka FYI. This looks like another case where the rules that get applied for source transformation don't handle things that class transformation does. I'm not asking you to do anything about this; just giving you an fyi so you have context if we ever attack this issue.

@scottmarlow I suggest you fix this in the existing code by loading the BeanManager and ValidatorFactory classes and checking the beginning of the FQCN of the loaded class. If it's 'jakarta' use that in the constants, otherwise use javax.  My assumption there is we want the constants to vary between jakarta and javax. :). If not what the Galleon plugin transformation is doing is wrong, and then the fix would be to break up those strings so they no longer match the transformation rule.

**Comment by scottmarlow** [comment]
Looks better now with your suggested changes @bstansberry! 
```
Stop processing [ /home/smarlow/.m2/repository/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ] using [ JAR ]: No changes$
[INFO] ================================================================================$
[INFO] [ Input  ] [ /home/smarlow/.m2/repository/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ]$
[INFO]            [ /home/smarlow/.m2/repository/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ]$
[INFO] [ Output ] [ /home/smarlow/.m2/repository/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ]$
[INFO]            [ /home/smarlow/work/wildfly/ee-9/feature-pack/target/jakarta-transform-maven-repo/org/wildfly/wildfly-jpa-jakarta/26.0.0.Beta1-SNAPSHOT/wildfly-jpa-jakarta-26.0.0.Beta1-SNAPSHOT.jar ]$
```

Thanks @yersan for helping with ^!

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
@scottmarlow My guess is the test failures relate to "jakarta" not being handled in https://github.com/wildfly/wildfly/blob/main/jpa/subsystem/src/main/java/org/jboss/as/jpa/classloader/TempClassLoader.java#L73

**Review by bstansberry — CHANGES_REQUESTED** [review]
Need to resolve test failures.

**Comment by ropalka** [comment]
Thanks @bstansberry for letting me know. I see @scottmarlow already fixed it. Yes, there are sometimes some tweaks needed to make things work.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks, @scottmarlow!

**Pull request merged**